platform.txt

```diff
@@ -70,9 +70,9 @@ compiler.elf2hex.extra_flags=
 
 ## generate file with git version number
 ## needs bash, git, and echo
-recipe.hooks.core.prebuild.1.pattern=bash -c "mkdir -p {build.path}/core && echo \#define ARDUINO_ESP8266_GIT_VER 0x`git --git-dir {runtime.platform.path}/.git rev-parse --short=8 HEAD 2>/dev/null || echo ffffffff` >{build.path}/core/core_version.h"
-## windows-compatible version may be added later
-recipe.hooks.core.prebuild.1.pattern.windows=
+recipe.hooks.core.prebuild.1.pattern=bash -c "mkdir -p {build.path}/core && echo \#define ARDUINO_ESP8266_GIT_VER `cd {runtime.platform.path}; git describe --tags 2>/dev/null || echo nix-{version}` >{build.path}/core/core_version.h"
```

Comment:
Can we keep `ARDUINO_ESP8266_GIT_VER` to be a short hash (expressed as an integer), and add a new define for the `git describe` output instead? E.g. `ARDUINO_ESP8266_GIT_DESC`. I have some release scripts which rely on ARDUINO_ESP8266_GIT_VER being what it is.

Reply:
Also, not necessarily `nix-` version. Also works on macOS, as far as i can tell.

Reply:
Will rename to GIT_DESC, and leave GIT_VER as before. About nix, it's a typo, unix was intended (for linux + macos).